spec/lib/chain-middleware.spec.js

```diff
@@ -0,0 +1,117 @@
+"use strict";
+
+describe("chainMiddleware", () => {
+    var chainMiddleware, fakeMiddlewareCalls, requestMock, responseMock;
+
+    /**
+     * Create fake middleware that just calls the next middleware
+     *
+     * @param {string} name
+     * @param {*} val to send to the `next` callback
+     * @return {Function}
+     */
+    function fakeMiddleware(name, val) {
+        return jasmine.createSpy(name).andCallFake((req, res, next) => {
+            fakeMiddlewareCalls.push(name);
+            next(val);
+        });
+    }
+
+    beforeEach(() => {
+        chainMiddleware = require("../../lib/chain-middleware");
+        requestMock = require("../mock/request-mock");
+        responseMock = require("../mock/response-mock");
```

Comment:
You don't need full mock objects for req and res.  These would work just fine because we never use `req` or `res` as objects.  They are simply passed verbatim to the middlewares.

    requestMock = "requestMock"
    responseMock = "responseMock"

If you do it this way, you are able to delete the two mock files as well.